packages/psss/src/api/api.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+import axios from 'axios';
+
+const PSSS_API_URL = process.env.REACT_APP_PSSS_API_URL;
+
+/*
+ * This module is responsible for pure api abstractions
+ */
+const headers = { 'X-Custom-Header': 'foobar' };
+const timeout = 45 * 1000;
+
+// withCredentials needs to be set for cookies to save @see https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials
+axios.defaults.withCredentials = true;
+
+const request = async (endpoint, options) => {
+  // Don't need as By default, axios serializes JavaScript objects to JSON
+  // if (body) {
+  //   fetchConfig.body =
+  //     typeof body === 'object' && !(body instanceof FormData) ? JSON.stringify(body) : body;
+  // }
+
+  try {
+    const { data } = await axios(`${PSSS_API_URL}/${endpoint}`, {
+      headers,
+      timeout,
+      ...options,
+    });
+    return data;
+  } catch (error) {
+    if (error.response) {
```

Comment:
I would expect data.error to always have the error message but I am also doing my best to always show a good error message. One thing I noticed is that all the errors from meditrak server come prepended with 'Internal server error'. For example 'Internal server error: Incorrect email or password'. I could parse it out but another option would be to not add it on the server?